client/state/notices/middleware.js

```diff
@@ -141,6 +146,36 @@ export const onPublicizeConnectionUpdateFailure = ( dispatch, { error } ) => dis
 	} ) )
 );
 
+export const onThemeUploadSuccess = ( dispatch, { themeId } ) => {
+	return dispatch( successNotice(
+		translate( 'Successfully uploaded theme %(name)s', {
+			args: {
+				name: themeId
```

Comment:
Yeah, I was actually considering even dropping the ID here since I figured the message would be clear enough without it, given UI context.